Problems when running continuous-integration/travis-ci/pr

The The Travis CI build is in progress shows errors and takes VERY (about 30 minutes) long:
```
...
 PASS  src/lib/__tests__/semver-test.js
 PASS  src/lib/__tests__/codeSign-test.js
 PASS  src/lib/__tests__/flowVersion-test.js
 PASS  src/lib/__tests__/cacheRepoUtils-test.js
 PASS  src/commands/__tests__/search-test.js
 PASS  src/lib/npm/__tests__/npmLibDefs-test.js
 PASS  src/lib/__tests__/libDefs-test.js
 PASS  src/commands/__tests__/install-test.js (6.411s)

  ● Console

    console.error src/lib/git.js:126

      { stderr: '\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email "you@example.com"\n  git config --global user.name "Your Name"\n\nto set your account\'s default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: empty ident name (for <travis@testing-docker-91fcb55b-2e84-4a9e-b20b-2e05f1bd3342.(none)>) not allowed\n',

        stdout: '',

        exitCode: 128 }

    console.error src/commands/install.js:463

        !! Failed to install underscore at /tmp/install-cmd-test-dir-844459/flowProj/flow-typed/npm/underscore_v1.x.x.js

    console.error src/commands/install.js:464

        ERROR: Error finding latest commit hash for definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js: undefined

Test Suites: 8 passed, 8 total
Tests:       94 passed, 94 total
Snapshots:   0 total
Time:        8.149s
Ran all test suites.
...
```

Now also problems in flow-typed itself on node 8:

```
src/lib/validationErrors.js -> dist/lib/validationErrors.js

> flow-typed@2.1.5 test-quick /home/travis/build/flowtype/flow-typed/cli

> jest && npm run lint

 PASS  src/lib/__tests__/semver-test.js

 PASS  src/lib/__tests__/codeSign-test.js

 PASS  src/lib/__tests__/flowVersion-test.js

 PASS  src/lib/__tests__/cacheRepoUtils-test.js

 PASS  src/commands/__tests__/search-test.js (5.807s)

 FAIL  src/lib/npm/__tests__/npmLibDefs-test.js (6.688s)

  ● npmLibDefs › extractLibDefsFromNpmPkgDir › succeeds on well-formed repo

    expect(received).toEqual(expected)

    

    Expected value to equal:

      [{"flowVersion": {"kind": "ranged", "lower": {"major": 0, "minor": 13, "patch": "x", "prerel": null}, "upper": {"major": 0, "minor": 37, "patch": "x", "prerel": null}}, "name": "underscore", "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.13.x-v0.37.x/underscore_v1.x.x.js", "scope": null, "testFilePaths": ["/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js"], "version": "v1.x.x"}, {"flowVersion": {"kind": "ranged", "lower": {"major": 0, "minor": 38, "patch": "x", "prerel": null}, "upper": null}, "name": "underscore", "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js", "scope": null, "testFilePaths": ["/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js", "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/test_underscore.js"], "version": "v1.x.x"}]

    Received:

      [{"flowVersion": {"kind": "ranged", "lower": {"major": 0, "minor": 38, "patch": "x", "prerel": null}, "upper": null}, "name": "underscore", "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js", "scope": null, "testFilePaths": ["/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js", "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/test_underscore.js"], "version": "v1.x.x"}, {"flowVersion": {"kind": "ranged", "lower": {"major": 0, "minor": 13, "patch": "x", "prerel": null}, "upper": {"major": 0, "minor": 37, "patch": "x", "prerel": null}}, "name": "underscore", "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.13.x-v0.37.x/underscore_v1.x.x.js", "scope": null, "testFilePaths": ["/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js"], "version": "v1.x.x"}]

    

    Difference:

    

    - Expected

    + Received

    

    @@ -2,45 +2,45 @@

       Object {

         "flowVersion": Object {

           "kind": "ranged",

           "lower": Object {

             "major": 0,

    -        "minor": 13,

    +        "minor": 38,

             "patch": "x",

             "prerel": null,

           },

    -      "upper": Object {

    -        "major": 0,

    -        "minor": 37,

    -        "patch": "x",

    -        "prerel": null,

    -      },

    +      "upper": null,

         },

         "name": "underscore",

    -    "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.13.x-v0.37.x/underscore_v1.x.x.js",

    +    "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js",

         "scope": null,

         "testFilePaths": Array [

           "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js",

    +      "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/test_underscore.js",

         ],

         "version": "v1.x.x",

       },

       Object {

         "flowVersion": Object {

           "kind": "ranged",

           "lower": Object {

             "major": 0,

    -        "minor": 38,

    +        "minor": 13,

             "patch": "x",

             "prerel": null,

           },

    -      "upper": null,

    +      "upper": Object {

    +        "major": 0,

    +        "minor": 37,

    +        "patch": "x",

    +        "prerel": null,

    +      },

         },

         "name": "underscore",

    -    "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js",

    +    "path": "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.13.x-v0.37.x/underscore_v1.x.x.js",

         "scope": null,

         "testFilePaths": Array [

           "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/test_underscore-v1.js",

    -      "/home/travis/build/flowtype/flow-typed/cli/src/lib/npm/__tests__/__npmLibDefs-fixtures__/extractLibDefsFromNpmPkgDir/well-formed/definitions/npm/underscore_v1.x.x/flow_v0.38.x-/test_underscore.js",

         ],

         "version": "v1.x.x",

       },

     ]

      

      at _callee$ (src/lib/npm/__tests__/npmLibDefs-test.js:40:28)

      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)

      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:355:22)

      at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:116:21)

      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)

      at invoke (node_modules/regenerator-runtime/runtime.js:154:20)

      at node_modules/regenerator-runtime/runtime.js:164:13

          at <anonymous>

 PASS  src/lib/__tests__/libDefs-test.js (7.171s)

 PASS  src/commands/__tests__/install-test.js (11.022s)

  ● Console

    console.error src/lib/git.js:126

      { stderr: '\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email "you@example.com"\n  git config --global user.name "Your Name"\n\nto set your account\'s default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: empty ident name (for <travis@testing-docker-fb0b2061-f531-448b-a1bb-cad424cee416.(none)>) not allowed\n',

        stdout: '',

        exitCode: 128 }

    console.error src/commands/install.js:463

        !! Failed to install underscore at /tmp/install-cmd-test-dir-842007/flowProj/flow-typed/npm/underscore_v1.x.x.js

    console.error src/commands/install.js:464

        ERROR: Error finding latest commit hash for definitions/npm/underscore_v1.x.x/flow_v0.38.x-/underscore_v1.x.x.js: undefined

Test Suites: 1 failed, 7 passed, 8 total

Tests:       1 failed, 93 passed, 94 total

Snapshots:   0 total

Time:        14.55s

Ran all test suites.

npm ERR! code ELIFECYCLE

npm ERR! errno 1

npm ERR! flow-typed@2.1.5 test-quick: `jest && npm run lint`

npm ERR! Exit status 1

npm ERR! 

npm ERR! Failed at the flow-typed@2.1.5 test-quick script.

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

npm ERR!     /home/travis/.npm/_logs/2017-07-20T10_49_48_963Z-debug.log

npm ERR! code ELIFECYCLE

npm ERR! errno 1

npm ERR! flow-typed@2.1.5 test: `npm run clean && npm run build && npm run test-quick`

npm ERR! Exit status 1

npm ERR! 

npm ERR! Failed at the flow-typed@2.1.5 test script.

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

npm ERR!     /home/travis/.npm/_logs/2017-07-20T10_49_48_995Z-debug.log

error Command failed with exit code 1.

info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

The command "./travis.sh" exited with 1.
cache.2

store build cache

0.01s

2.10snothing changed, not updating cache

Done. Your build exited with 1.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,6 +12,9 @@ addons:
     packages:
       - unzip
       - g++-4.8
+before_install:
+  - git config --global user.name "Travis CI"
+  - git config --global user.email "travis@flowtyped.com"
 script: ./travis.sh
 cache:
   directories:

